src/ElCamino.AspNetCore.Identity.AzureTable/IdentityCloudContext.cs

```diff
@@ -28,6 +28,12 @@ protected virtual void Initialize(IdentityConfiguration config)
             _config = config;
             _client = new TableServiceClient(_config.StorageConnectionString);
 
+            if (config.TokenCredential != null)

```

Comment:
Probably need to check for the _config.TokenCredential first since the line 29 will likely throw an exception in the _config.StorageConnectionString isNullOrEmpty. Maybe even throw an argument exception if both _config.TokenCredential  and _config.StorageConnectionString are both null.

Reply:
You still need to provide some form of connection string (but the Access Key is ignored) so I've left that bit as is.

The alternative is that _instead_ of a connection string you would need to collect the URL of the endpoint

Happy to put in an ArgumentException for StorageConnectionString, but as there wasn't one already I left it as is

Reply:
Btw - thanks for the feature. This will be a nice option to be able to support managed identities.

Reply:
Sorry for the delay in coming back to this. The way I'd written it (and this may not make much sense, but it's the way that we've been using these objects) is a bit unexpected:

When you're creating a `TableServiceClient` with a TokenCredential, you need the full URL of the table service that you're connecting to. Instead of creating a new variable for that, I was re-using the value from within the connection string. This means that you would need to pass both a `StorageConnectionString` _and_ a `TokenCredential` in to use managed identities, and the Key part of the connection string would be ignore.

That's why I create one instance of the client first (because there's no other way to parse the connection string) and then obtain the URL from that.

Reply:
Hi Folks, has this gone anywhere? IMHO, it's absolutely essential to have Managed Identity authentication these days. We just need the highest level of security without exception. I'll go ahead and adapt this for the newest version, and raise another pull request.